python/smqtk/representation/key_value/postgres.py

```diff
@@ -22,8 +22,15 @@
 class PostgresKeyValueStore (KeyValueStore):
     """
     PostgreSQL-backed key-value storage.
+
+    NOTE: Due to the differences in pickle serialization between python
+    versions 2 and 3 (even with the same protocol version) the same underlying
+    postgresql table should not be shared between instances of
+    ``PostgresKeyValueStore`` in different python major versions.
     """
 
+    PICKLE_PROTOCOL_VER = -1
```

Comment:
Will Python 3 attempt to read objects that have been pickled in Python 2? The [documentation](https://docs.python.org/3/library/pickle.html#pickle.load) makes it seem like this is the case, which would defeat the purpose of changing the protocol number.

Reply:
A) I just noticed that I didn't try using the ``fix_imports`` flag for python 3's pickle. I'll have to try that in my compatibility failures I was seeing.

B) A python 3 instance of ``PostgresKeyValueStore`` will attempt to read the data in the table and columns its pointed to, even if that data was written by a python 2 instance of ``PostgresKeyValueStore``.  I had tried using a pinned protocol version of 2 (since both python versions understand that version) but was still seeing compatibility issues.  This may have been the correct thing to do when I try (A).